### PR TITLE
TYP: Annotated pandas/core/indexing.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -273,10 +273,10 @@ def _convert_slice_indexer(self, key: slice, axis: int):
         ax = self.obj._get_axis(min(axis, self.ndim - 1))
         return ax._convert_slice_indexer(key, kind=self.name)
 
-    def _has_valid_setitem_indexer(self, indexer):
+    def _has_valid_setitem_indexer(self, indexer) -> bool:
         return True
 
-    def _has_valid_positional_setitem_indexer(self, indexer):
+    def _has_valid_positional_setitem_indexer(self, indexer) -> bool:
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: it looks like this always either raises or returns True; is that accurate?  if so, maybe get rid of the return value altogether (might need to update usages)

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -2523,7 +2523,7 @@ def _maybe_numeric_slice(df, slice_, include_bool=False):
     return slice_
 
 
-def _can_do_equal_len(labels, value, plane_indexer, lplane_indexer, obj):
+def _can_do_equal_len(labels, value, plane_indexer, lplane_indexer, obj) -> bool:
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: can any of the args here be typed?

File: pandas/core/indexing.py [hunk 3]
```diff
@@ -2465,12 +2465,12 @@ def is_nested_tuple(tup, labels):
...
 
 
-def is_label_like(key):
+def is_label_like(key) -> bool:
     # select a label or row
     return not isinstance(key, slice) and not is_list_like_indexer(key)
 
 
-def need_slice(obj):
+def need_slice(obj) -> bool:
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: seems like obj will be a slice?

File: pandas/core/indexing.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/core/indexing.py]: I don't feel confident enough in reformatting this function, I'll leave it for someone else who will probably won't break the package.

**Review by WillAyd — APPROVED** [review]
lgtm. I think additional types as proposed by @jbrockmendel would be nice if not too much extra effort

**Comment by MomIsBestFriend** [comment]
> lgtm. I think additional types as proposed by @jbrockmendel would be nice if not too much extra effort

The problem is that I don't know how to annotate them:(

**Comment by WillAyd** [comment]
OK thanks @MomIsBestFriend . Sure can always add more types as you get more familiar with code base

**Pull request merged**